fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneHDFSScanColumnRule.java

```diff
@@ -31,6 +31,7 @@
 public class PruneHDFSScanColumnRule extends TransformationRule {
     public static final PruneHDFSScanColumnRule HIVE_SCAN = new PruneHDFSScanColumnRule(OperatorType.LOGICAL_HIVE_SCAN);
     public static final PruneHDFSScanColumnRule ICEBERG_SCAN = new PruneHDFSScanColumnRule(OperatorType.LOGICAL_ICEBERG_SCAN);
+    public static final PruneHDFSScanColumnRule HUDI_SCAN = new PruneHDFSScanColumnRule(OperatorType.LOGICAL_HUDI_SCAN);
```

Comment:
this will throw "Unsupported logical scan operator type" since we do not add implementation for HUDI

Reply:
rebase conflicts here, revised.